capella2polarion/converters/converter_config.py

```diff
@@ -39,20 +43,98 @@ class LinkConfig:
     link_field: str = ""
     reverse_field: str = ""
 
+    @staticmethod
+    def _force_config(
+        links: list[str | dict[str, t.Any]], role_prefix: str = ""
+    ) -> list[LinkConfig]:
+        result: list[LinkConfig] = []
+        for link in links:
+            if isinstance(link, str):
+                config = LinkConfig(
+                    capella_attr=link,
+                    polarion_role=add_prefix(link, role_prefix),
+                    link_field=link,
+                    reverse_field=f"{link}_reverse",
+                )
+            elif isinstance(link, dict):
+                config = LinkConfig(
+                    capella_attr=(lid := link["capella_attr"]),
+                    polarion_role=add_prefix(
+                        (pid := link.get("polarion_role", lid)),
+                        role_prefix,
+                    ),
+                    include=link.get("include", {}),
+                    link_field=(lf := link.get("link_field", pid)),
+                    reverse_field=link.get("reverse_field", f"{lf}_reverse"),
+                )
+            else:
+                logger.error(  # type: ignore[unreachable]
+                    "Link not configured correctly: %r",
+                    link,
+                )
+                continue
+            result.append(config)
+        return result
+
 
 @dataclasses.dataclass
 class CapellaTypeConfig:
     """A single Capella Type configuration."""
 
     p_type: str | None = None
-    converters: str | list[str] | dict[str, dict[str, t.Any]] | None = None
+    converters: dict[str, dict[str, t.Any]] | None = None
     links: list[LinkConfig] = dataclasses.field(default_factory=list)
     is_actor: bool | None = None
     nature: str | None = None
 
-    def __post_init__(self):
-        """Post processing for the initialization."""
-        self.converters = _force_dict(self.converters)
+    @staticmethod
+    def _force_dict(
+        converters: str | list[str] | ConvertersType | None,
+    ) -> dict[str, dict[str, t.Any]]:
```

Comment:
This method is marked as protected, but only accessed from outside the class it is defined in. For me it would make way more sense to place it in the `ConverterConfig` class. The same applies to the `_filter_config` method which is only called by this method